pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,296 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shoot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/operation"
+	"github.com/gardener/gardener/pkg/operation/common"
+	utilerrors "github.com/gardener/gardener/pkg/utils/errors"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/version"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	botanistpkg "github.com/gardener/gardener/pkg/operation/botanist"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	retryutils "github.com/gardener/gardener/pkg/utils/retry"
+)
+
+func (c *Controller) prepareShootForMigration(shoot *gardencorev1beta1.Shoot, logger *logrus.Entry) (reconcile.Result, error) {
+
+	var (
+		operationType              = gardencorev1beta1.LastOperationTypeMigrate
+		respectSyncPeriodOverwrite = c.respectSyncPeriodOverwrite()
+		failed                     = common.IsShootFailed(shoot)
+		ignored                    = common.ShouldIgnoreShoot(respectSyncPeriodOverwrite, shoot)
+	)
+
+	if shoot.Status.SeedName == nil {
+		err := fmt.Errorf("missing Seed's name in Shoot's status")
+		return reconcile.Result{}, utilerrors.WithSuppressed(err, c.updateShootStatusError(shoot, fmt.Sprintf("Can't prepare Shoot's control plane for migration: %s", err.Error())))
+	}
+
+	if failed || ignored {
+		return reconcile.Result{}, fmt.Errorf("Shoot %s is failed or ignored, will skip migration preparation", shoot.GetName())
+	}
+
+	targetSeedAvailable := c.isSeedAvailable(shoot.Spec.SeedName, logger)
+	if !targetSeedAvailable {
+		return reconcile.Result{}, fmt.Errorf("Target Seed is not available to host the Shoot's %s Control Plane", shoot.GetName())
+	}
+
+	o, err := operation.New(shoot, shoot.Status.SeedName, c.config, logger, c.k8sGardenClient, c.k8sGardenCoreInformers.Core().V1beta1(), c.identity, c.secrets, c.imageVector)
+	if err != nil {
+		return reconcile.Result{}, utilerrors.WithSuppressed(err, c.updateShootStatusError(shoot, fmt.Sprintf("Could not initialize a new operation for Shoot control plane preparation for migration: %s", err.Error())))
+	}
+
+	if o.Seed.Info.Spec.Backup == nil {
+		return reconcile.Result{}, fmt.Errorf("Seed is not configured for backup")
```

Comment:
```suggestion
		return reconcile.Result{}, fmt.Errorf("seed is not configured for backup")
```

I think it would make sense to add an Event here stating that the migration from on this is not possible.